skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+        self._interval_metadata._intervals.append(self)
+
+    @experimental(as_of='0.5.0-dev')
+    def __eq__(self, other):
+        '''Test if this ``Interval`` object is equal to another.
+
+        The equality is performed by checking if the ``metadata``,
+        ``location`` and ``boundaries`` are equal.
+
+        Parameters
+        ----------
+        other : Interval
+            Interval to test for equality against.
+
+        Returns
+        -------
+        bool
+            Indicates if the two objects are equal.
+        '''
+        return ((self.metadata == other.metadata) and
+                (self.locations == other.locations) and
+                (self.boundaries == other.boundaries))
+
+    @experimental(as_of='0.5.0-dev')
+    def __ne__(self, other):
+        '''Test if this ``Interval`` object is not equal to another.
+
+        Parameters
+        ----------
+        other : Interval
+            Interval to test for inequality against.
+
+        Returns
+        -------
+        bool
+            Indicates if the two objects are not equal.
+        '''
+        return not self.__eq__(other)
+
+    @experimental(as_of='0.5.0-dev')
+    def __repr__(self):
+        '''Return a string representation of this ``Interval`` object.
+
+        Returns
+        -------
+        str
+            String representation of this ``Interval`` object.
+        '''
+        s = ('{}(interval_metadata=<{!r}>, locations={!r}, '
+             'boundaries={!r}, metadata={!r})')
+        return s.format(self.__class__.__name__, id(self._interval_metadata),
+                        self.locations, self.boundaries, self.metadata)
+
+    @experimental(as_of='0.5.0-dev')
+    def drop(self):
+        '''Drop this ``Interval`` object from the interval metadata it links to.
+
+        See Also
+        --------
+        skbio.metadata.IntervalMetadata.drop
+        '''
+        self._interval_metadata.drop(locations=self.locations,
+                                     metadata=self.metadata)
+        self._interval_metadata = None
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def boundaries(self):
+        '''The openness of each coordinate.
+
+        This indicates that the exact boundary point of a interval feature
+        is unknown. The location may begin or end at some points outside
+        the specified coordinates. This accommodates the location format [1]_
+        of INSDC.
+
+        References
+        ----------
+        .. [1] ftp://ftp.ebi.ac.uk/pub/databases/embl/doc/FT_current.html#3.4.3
+        '''
+        if self.dropped:
+            raise RuntimeError('Cannot retrieve boundaries from.'
+                               'dropped Interval object.')
+        return self._boundaries
+
+    @boundaries.setter
+    @experimental(as_of='0.5.0-dev')
+    def boundaries(self, value):
+        '''Set ``boundaries``.
+
+        The ``value`` should be ``None`` (meaning all boundaries are ``True``)
```

Comment:
Sorry I wasn't very clear about handling `None`.

I think it makes sense to disallow `None` here and have the deleter (which doesn't accept any input) reset `boundaries` to all `Trues`, since this is the default `boundaries` representation and you'd get this behavior in the constructor if you didn't provide boundaries. This idea is analogous to the `metadata` deleter resetting `metadata` to an empty `dict` (its default representation). Does that make sense?